### ISSUE TITLE
How to build Bolero without Paket?

### ISSUE THREAD

**Issue by xperiandri** [fsbolero/Bolero#33]
I don't use Paket in my solution so I would like to build Bolero without it.
But client-side build breaks
```
1>------ Build started: Project: Admin, Configuration: Debug Any CPU ------
1>Admin -> C:\Users\Andrii\Dev\SFSync\SalesforceTopicsListener\src\Admin\bin\Debug\netstandard2.0\Admin.dll
1>Cannot find declaration of exported type 'System.Threading.Semaphore' from the assembly 'System.Threading, Version=4.0.12.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
1>Cannot find declaration of exported type 'System.AggregateException' from the assembly 'FSharp.Core, Version=4.6.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
1>Cannot find declaration of exported type 'System.Threading.CancellationToken' from the assembly 'FSharp.Core, Version=4.6.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
1>Cannot find declaration of exported type 'System.Threading.CancellationTokenRegistration' from the assembly 'FSharp.Core, Version=4.6.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
1>Cannot find declaration of exported type 'System.Threading.CancellationTokenSource' from the assembly 'FSharp.Core, Version=4.6.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
1>Processing embedded resource linker descriptor: mscorlib.xml
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Threading.ExecutionContext (Nothing).  Duplicate uses (Nothing)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Threading.ExecutionContext System.Threading.ExecutionContext::Capture()
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Threading.ExecutionContext System.Threading.ExecutionContext::Capture(System.Threading.StackCrawlMark&,System.Threading.ExecutionContext/CaptureOptions)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Threading.Interlocked (Nothing).  Duplicate uses (Nothing)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Int32 System.Threading.Interlocked::CompareExchange(System.Int32&,System.Int32,System.Int32)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Int32 System.Threading.Interlocked::CompareExchange(System.Int32&,System.Int32,System.Int32,System.Boolean&)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Object System.Threading.Interlocked::CompareExchange(System.Object&,System.Object,System.Object)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Single System.Threading.Interlocked::CompareExchange(System.Single&,System.Single,System.Single)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Int64 System.Threading.Interlocked::CompareExchange(System.Int64&,System.Int64,System.Int64)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.IntPtr System.Threading.Interlocked::CompareExchange(System.IntPtr&,System.IntPtr,System.IntPtr)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Double System.Threading.Interlocked::CompareExchange(System.Double&,System.Double,System.Double)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of T System.Threading.Interlocked::CompareExchange(T&,T,T)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Int32 System.Threading.Interlocked::Exchange(System.Int32&,System.Int32)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Object System.Threading.Interlocked::Exchange(System.Object&,System.Object)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Single System.Threading.Interlocked::Exchange(System.Single&,System.Single)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Int64 System.Threading.Interlocked::Exchange(System.Int64&,System.Int64)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.IntPtr System.Threading.Interlocked::Exchange(System.IntPtr&,System.IntPtr)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Double System.Threading.Interlocked::Exchange(System.Double&,System.Double)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of T System.Threading.Interlocked::Exchange(T&,T)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Threading.Monitor (Nothing).  Duplicate uses (Nothing)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Void System.Threading.Monitor::Enter(System.Object)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Void System.Threading.Monitor::Enter(System.Object,System.Boolean&)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Void System.Threading.Monitor::Exit(System.Object)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Threading.ManualResetEvent (Nothing).  Duplicate uses (Nothing)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Void System.Threading.ManualResetEvent::.ctor(System.Boolean)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Threading.SynchronizationLockException (Nothing).  Duplicate uses (Nothing)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Void System.Threading.SynchronizationLockException::.ctor(System.String)
1>Duplicate preserve in resource mscorlib.xml in mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e of System.Threading.WasmRuntime (All).  Duplicate uses (All)
1>Fatal error in IL Linker
1>Mono.Linker.MarkException: Error processing method: 'System.Void Bolero.Templating.Client.SignalRClient::.ctor(Bolero.Templating.HotReloadSettings)' in assembly: 'Bolero.Templating.dll' ---> Mono.Cecil.ResolutionException: Failed to resolve System.Void Blazor.Extensions.HubConnection::OnClose(System.Func`2<System.Exception,System.Threading.Tasks.Task>)
1>   at Mono.Linker.Steps.MarkStep.HandleUnresolvedMethod(MethodReference reference)
1>   at Mono.Linker.Steps.MarkStep.MarkMethod(MethodReference reference)
1>   at Mono.Linker.Steps.MarkStep.MarkInstruction(Instruction instruction)
1>   at Mono.Linker.Steps.MarkStep.MarkMethodBody(MethodBody body)
1>   at Mono.Linker.Steps.MarkStep.ProcessMethod(MethodDefinition method)
1>   at Mono.Linker.Steps.MarkStep.ProcessQueue()
1>   --- End of inner exception stack trace ---
1>   at Mono.Linker.Steps.MarkStep.ProcessQueue()
1>   at Mono.Linker.Steps.MarkStep.ProcessPrimaryQueue()
1>   at Mono.Linker.Steps.MarkStep.Process()
1>   at Mono.Linker.Steps.MarkStep.Process(LinkContext context)
1>   at Mono.Linker.Pipeline.ProcessStep(LinkContext context, IStep step)
1>   at Mono.Linker.Pipeline.Process(LinkContext context)
1>   at Mono.Linker.Driver.Run(ILogger customLogger)
1>   at Mono.Linker.Driver.Execute(String[] args, ILogger customLogger)
1>C:\Users\Andrii\.nuget\packages\microsoft.aspnetcore.blazor.build\0.7.0\targets\Blazor.MonoRuntime.targets(439,5): error MSB3073: The command "dotnet "C:\Users\Andrii\.nuget\packages\microsoft.aspnetcore.blazor.build\0.7.0\targets\../tools/illink/illink.dll" -l none --verbose --strip-security true --exclude-feature com --exclude-feature sre -v false -c link -u link -b true -d "C:\Users\Andrii\.nuget\packages\microsoft.aspnetcore.blazor.build\0.7.0\targets\../tools/mono/bcl/" -d "C:\Users\Andrii\.nuget\packages\microsoft.aspnetcore.blazor.build\0.7.0\targets\../tools/mono/bcl/Facades/" -o "C:\Users\Andrii\Dev\SFSync\SalesforceTopicsListener\src\Admin\obj\Debug\netstandard2.0\blazor\linker/" -x "C:\Users\Andrii\.nuget\packages\microsoft.aspnetcore.blazor.build\0.7.0\targets\BuiltInBclLinkerDescriptor.xml" -x "C:\Users\Andrii\Dev\SFSync\SalesforceTopicsListener\src\Admin\obj\Debug\netstandard2.0\blazor\linker.descriptor.xml" -a "C:\Users\Andrii\.nuget\packages\blazor.extensions.signalr\0.1.0\lib\netstandard2.0\Blazor.Extensions.SignalR.JS.dll" -a "C:\Users\Andrii\.nuget\packages\blazor.extensions.signalr\0.1.0\lib\netstandard2.0\Blazor.Extensions.SignalR.dll" -a "C:\Users\Andrii\.nuget\packages\bolero\0.3.130\lib\netstandard2.0\Bolero.Server.dll" -a "C:\Users\Andrii\.nuget\packages\bolero\0.3.130\lib\netstandard2.0\Bolero.dll" -a "C:\Users\Andrii\.nuget\packages\bolero.hotreload\0.3.130\lib\netstandard2.0\Bolero.Templating.dll" -a "C:\Users\Andrii\.nuget\packages\elmish\2.0.0\lib\netstandard2.0\Elmish.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\FSharp.Core.dll" -a "C:\Users\Andrii\.nuget\packages\htmlagilitypack\1.8.0\lib\netstandard2.0\HtmlAgilityPack.dll" -a "C:\Users\Andrii\.nuget\packages\microsoft.aspnetcore.blazor\0.7.0\lib\netstandard2.0\Microsoft.AspNetCore.Blazor.dll" -a "C:\Users\Andrii\.nuget\packages\microsoft.aspnetcore.blazor.browser\0.7.0\lib\netstandard2.0\Microsoft.AspNetCore.Blazor.Browser.dll" -a "C:\Users\Andrii\.nuget\packages\microsoft.aspnetcore.blazor.build\0.7.0\lib\netstandard1.0\Microsoft.AspNetCore.Blazor.TagHelperWorkaround.dll" -a "C:\Users\Andrii\.nuget\packages\microsoft.aspnetcore.http.abstractions\2.1.0\lib\netstandard2.0\Microsoft.AspNetCore.Http.Abstractions.dll" -a "C:\Users\Andrii\.nuget\packages\microsoft.aspnetcore.http.features\2.1.0\lib\netstandard2.0\Microsoft.AspNetCore.Http.Features.dll" -a "C:\Users\Andrii\.nuget\packages\microsoft.extensions.dependencyinjection\2.1.0\lib\netstandard2.0\Microsoft.Extensions.DependencyInjection.dll" -a "C:\Users\Andrii\.nuget\packages\microsoft.extensions.dependencyinjection.abstractions\2.1.0\lib\netstandard2.0\Microsoft.Extensions.DependencyInjection.Abstractions.dll" -a "C:\Users\Andrii\.nuget\packages\microsoft.extensions.primitives\2.1.0\lib\netstandard2.0\Microsoft.Extensions.Primitives.dll" -a "C:\Users\Andrii\.nuget\packages\microsoft.jsinterop\0.7.0\lib\netstandard2.0\Microsoft.JSInterop.dll" -a "C:\Users\Andrii\.nuget\packages\mono.webassembly.interop\0.7.0\lib\netstandard2.0\Mono.WebAssembly.Interop.dll" -a "C:\Users\Andrii\.nuget\packages\system.buffers\4.4.0\lib\netstandard2.0\System.Buffers.dll" -a "C:\Users\Andrii\.nuget\packages\system.collections.concurrent\4.3.0\lib\netstandard1.3\System.Collections.Concurrent.dll" -a "C:\Users\Andrii\.nuget\packages\system.diagnostics.diagnosticsource\4.3.0\lib\netstandard1.3\System.Diagnostics.DiagnosticSource.dll" -a "C:\Users\Andrii\.nuget\packages\system.io.filesystem.primitives\4.3.0\lib\netstandard1.3\System.IO.FileSystem.Primitives.dll" -a "C:\Users\Andrii\.nuget\packages\system.linq\4.3.0\lib\netstandard1.6\System.Linq.dll" -a "C:\Users\Andrii\.nuget\packages\system.memory\4.5.0\lib\netstandard2.0\System.Memory.dll" -a "C:\Users\Andrii\.nuget\packages\system.numerics.vectors\4.4.0\lib\netstandard2.0\System.Numerics.Vectors.dll" -a "C:\Users\Andrii\.nuget\packages\system.runtime.compilerservices.unsafe\4.5.0\lib\netstandard2.0\System.Runtime.CompilerServices.Unsafe.dll" -a "C:\Users\Andrii\.nuget\packages\system.runtime.numerics\4.3.0\lib\netstandard1.3\System.Runtime.Numerics.dll" -a "C:\Users\Andrii\.nuget\packages\system.security.cryptography.openssl\4.3.0\lib\netstandard1.6\System.Security.Cryptography.OpenSsl.dll" -a "C:\Users\Andrii\.nuget\packages\system.security.cryptography.primitives\4.3.0\lib\netstandard1.3\System.Security.Cryptography.Primitives.dll" -a "C:\Users\Andrii\.nuget\packages\system.text.encodings.web\4.5.0\lib\netstandard2.0\System.Text.Encodings.Web.dll" -a "C:\Users\Andrii\.nuget\packages\system.text.regularexpressions\4.3.0\lib\netstandard1.6\System.Text.RegularExpressions.dll" -a "C:\Users\Andrii\.nuget\packages\system.threading\4.3.0\lib\netstandard1.3\System.Threading.dll" -a "C:\Users\Andrii\.nuget\packages\system.threading.tasks.extensions\4.3.0\lib\netstandard1.0\System.Threading.Tasks.Extensions.dll" -a "C:\Users\Andrii\.nuget\packages\system.xml.readerwriter\4.3.0\lib\netstandard1.3\System.Xml.ReaderWriter.dll" -a "C:\Users\Andrii\.nuget\packages\system.xml.xmldocument\4.3.0\lib\netstandard1.3\System.Xml.XmlDocument.dll" -a "C:\Users\Andrii\.nuget\packages\system.xml.xpath\4.3.0\lib\netstandard1.3\System.Xml.XPath.dll" -a "C:\Users\Andrii\.nuget\packages\system.xml.xpath.xmldocument\4.3.0\lib\netstandard1.3\System.Xml.XPath.XmlDocument.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\cs\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\de\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\en\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\es\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\fr\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\it\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\ja\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\ko\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\pl\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\pt-BR\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\ru\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\tr\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\zh-Hans\FSharp.Core.resources.dll" -a "C:\Users\Andrii\.nuget\packages\fsharp.core\4.6.2\lib\netstandard1.6\zh-Hant\FSharp.Core.resources.dll" -a "C:\Users\Andrii\Dev\SFSync\SalesforceTopicsListener\src\Admin\obj\Debug\netstandard2.0\Admin.dll"" exited with code 1.
1>Done building project "Admin.fsproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

``` XML
<?xml version="1.0" encoding="utf-8"?>
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <RunCommand>dotnet</RunCommand>
    <RunArguments>blazor serve</RunArguments>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="Main.fs" />
    <Compile Include="Startup.fs" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Bolero.Build" Version="0.3.130" />
    <PackageReference Include="Bolero.HotReload" Version="0.3.130" />
    <PackageReference Include="Microsoft.AspNetCore.Blazor.Build" Version="0.7.0" />
    <!--<PackageReference Include="Microsoft.AspNetCore.Blazor.Cli" Version="0.7.0" />-->
  </ItemGroup>

</Project>
```

**Comment by Tarmil** [comment]
Okay, I found that this is because we have a dependency on Blazor.Extensions.SignalR with a version that is too loose. Paket takes the highest compatible version, which works, but Nuget takes the lowest compatible version, which doesn't work. I'll fix the dependency in the upcoming Bolero 0.4. In the meantime you can add the following to your project:

```xml
    <PackageReference Include="Blazor.Extensions.SignalR" Version="0.1.8" />
```

**Comment by xperiandri** [comment]
@Tarmil thanks, your workaround works
